pkg/querier/engine/streaming/operator/range_vector_selector_with_transformation.go

```diff
@@ -0,0 +1,197 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+// Provenance-includes-location: https://github.com/prometheus/prometheus/blob/main/promql/engine.go
+// Provenance-includes-location: https://github.com/prometheus/prometheus/blob/main/promql/functions.go
+// Provenance-includes-license: Apache-2.0
+// Provenance-includes-copyright: The Prometheus Authors
+
+package operator
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/model/value"
+	"github.com/prometheus/prometheus/promql"
+	"github.com/prometheus/prometheus/tsdb/chunkenc"
+)
+
+type RangeVectorSelectorWithTransformation struct {
```

Comment:
[nit] Could you add a description to this struct? I found it a bit hard to understand what it does and name didn't help to me.

Reply:
Good point, added a comment in 10bf1cd